integration/graphql/src/main/java/com/blazebit/persistence/integration/graphql/GraphQLEntityViewSupport.java

```diff
@@ -624,50 +602,41 @@ public void applyFetches(DataFetchingEnvironment dataFetchingEnvironment, Entity
      * @param elementRoot The element root
      */
     public void applyFetches(DataFetchingEnvironment dataFetchingEnvironment, EntityViewSetting<?, ?> setting, String elementRoot) {
-        String prefix = elementRoot == null || elementRoot.isEmpty() ? "" : elementRoot + "/";
-        StringBuilder sb = new StringBuilder();
         DataFetchingFieldSelectionSet selectionSet = dataFetchingEnvironment.getSelectionSet();
-        GraphQLFieldsContainer rootType = (GraphQLFieldsContainer) getElementType(dataFetchingEnvironment, elementRoot);
-        try {
-            OUTER:
-            for (SelectedField field : selectionSet.getFields()) {
-                String key = field.getQualifiedName();
-                if (key.length() < prefix.length()) {
+        OUTER:
+        for (SelectedField field : selectionSet.getFields()) {
+            if (!isLeaf(field.getType())) {
+                // Skip non-leaf types because these intermediate objects types would
+                // lead to fetching the whole subtree in EntityViewConfiguration#getFetches
+                continue;
+            }
+            String[] fieldParts = field.getFullyQualifiedName().split("/");
```

Comment:
Is that the new format? Fields separated by `/` and every field has the format `TYPE_NAME.FIELD_NAME`? This graphql-java runtime surely is breaking things all the time..

Anyway, I think we should add caching for the fetch resolving. AFAIU, since EVs forbid cycles, there should only be a limited set of field paths possible, and since the paths are now apparently typed, we can use the FQN as cache key. Can you please add a `ConcurrentHashMap` as a cache for this?

Reply:
Ah, just made the same suggestion in the other comment :-)

Reply:
> This graphql-java runtime surely is breaking things all the time...

To be fair, this time they did not break the contract. `getFullyQualifiedName` is a new method.

Reply:
> Can you please add a ConcurrentHashMap as a cache for this?

You mentioning ConcurrentHashMap suggests that you want to make the cache's lifecycle tied to the app lifecycle. I originally dismissed this thought from the start because you would have to somehow do cache pruning (see my comment above about request-only cache) but thinking about it more, the number of field paths in the schema is limited by definition, although it can be quite large when taking all combinations. Perhaps it would suffice to just cache the mapping for each `SelectedField` and not also all its subfields/subpaths? Then the cache size would be limited by the number of leaf nodes in the schema which should certainly be manageable.

WDYT?

Reply:
I think we should aim for a single big cache for the whole app and add a config option to disable the caching.

Reply:
Ok, will do that. Can I consider the other discussions regarding memory allocations as superseded by this?